app/models/procedure_revision.rb

```diff
@@ -181,6 +181,16 @@ def children_of(tdc)
     end
   end
 
+  def parent_of(tdc)
+    revision_types_de_champ
+      .find { _1.type_de_champ_id == tdc.id }&.parent&.type_de_champ
+  end
+
+  def child?(tdc)
+    revision_types_de_champ
+      .find { _1.type_de_champ_id == tdc.id }&.child?
```

Comment:
maybe

```suggestion
      .find { _1.type_de_champ_id == tdc.id }&.child? || false
```

Aussi est-ce que sur ces méthodes on pourrait vraiment ne pas trouver le rtdc correspondant ?

Reply:
J'ai écrit un code un peu trop défensif. Partons du principe que c'est effectivement impossible.